indigo_app/views/works.py

```diff
@@ -559,6 +567,47 @@ def get_works(self, table):
 
         return works
 
+    def get_tasks(self, works, form):
+        tasks = []
+
+        def make_task():
+            task = Task()
+            task.country = self.country
+            task.locality = self.locality
+            task.created_by_user = self.request.user
+            task.title = chosen_task.capitalize()
+            for possible_task in form.possible_tasks:
+                if chosen_task == possible_task['key']:
+                    task.description = possible_task['description']
+
+            # need to save before assigning work because of M2M relation
+            task.save()
+            task.work = work.get('work')
+            task.save()
+            tasks.append(task)
+
+        # bulk create tasks on primary works
+        if form.cleaned_data.get('primary_tasks'):
+            for work in works:
+                if work['status'] == 'success' and not work['work'].stub:
+                    for chosen_task in form.cleaned_data.get('primary_tasks'):
+                        make_task()
+
+        # bulk create tasks on all works
+        if form.cleaned_data.get('all_tasks'):
+            for work in works:
+                if work['status'] == 'success':
+                    for chosen_task in form.cleaned_data.get('all_tasks'):
+                        make_task()
+
+        # add all the tasks to the workflows selected
+        if form.cleaned_data.get('workflows'):
+            for workflow in form.cleaned_data.get('workflows'):
+                workflow.tasks = tasks
```

Comment:
This will make these tasks the ONLY tasks in this workflow. You probably want to do `workflow.tasks.add(*tasks)`

Reply:
yeah this makes way more sense thanks – lmk what you think of my change

Reply:
Looks good. Just check `.all()` works when assigning the workflows to the tasks.